collect_app/src/test/java/org/odk/collect/android/external/FormUriActivityTest.kt

```diff
@@ -125,6 +140,14 @@ class FormUriActivityTest {
             override fun providesScheduler(workManager: WorkManager?): Scheduler {
                 return fakeScheduler
             }
+
+            override fun providesSavepointFinder(): SavepointFinder {
```

Comment:
I think you can actually just treat `savepointsRepository` as the "boundary" for the test here instead of needing to inject a stub `SavepointFinder`. You don't need to switch out both the repository and the "use case". In that case, you can just save the test `SavePoint` instances to the in mem repo. For example:

```kotlin
@Test
fun `If there is a savepoint, display a recovery dialog before starting a blank form`() {
    val project = Project.Saved("123", "First project", "A", "#cccccc")
    projectsRepository.save(project)
    whenever(projectsDataService.getCurrentProject()).thenReturn(project)

    val form = formsRepository.save(
        FormUtils.buildForm(
            "1",
            "1",
            TempFiles.createTempDir().absolutePath
        ).build()
    )

    val savepointFile = TempFiles.createTempFile()
    val savepoint = Savepoint(form.dbId, null, savepointFile.absolutePath, TempFiles.createTempFile().absolutePath)
    savepointsRepository.save(savepoint)

    launcherRule.launch<FormUriActivity>(getBlankFormIntent(project.uuid, form.dbId))
    fakeScheduler.flush()

    assertSavepointRecoveryDialog(savepointFile)
}
```

Reply:
Done.